### PR TITLE
FLPATH-421: persist user at login

### PR DESCRIPTION
**Pull request by RichardW98**
**What this PR does / why we need it**:
persist authenticated user in db
**Which issue(s) this PR fixes** 
Fixes #421

**Change type**
- [ ] New feature
- [ ] Bug fix
- [ ] Unit tests
- [ ] Integration tests
- [ ] CI
- [ ] Documentation
- [ ] Auto-generated SDK code

**Impacted services**
- [ ] Workflow Service
- [ ] Notification Service

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.

### CODE REVIEW

File: workflow-service/src/main/java/com/redhat/parodos/security/LoginController.java [hunk 1]
```diff
@@ -39,13 +44,25 @@
...
+	public LoginController(UserService userService) {
+		this.userService = userService;
+	}
+
 	@Operation(summary = "Login")
 	@ApiResponses(value = { @ApiResponse(responseCode = "200", description = "Succeeded", content = @Content),
 			@ApiResponse(responseCode = "401", description = "Unauthorized", content = @Content) })
 	@GetMapping
 	public ResponseEntity<String> login() {
+		String username = SecurityUtils.getUsername();
```
Comment (anludke) [inline workflow-service/src/main/java/com/redhat/parodos/security/LoginController.java:61]: Maybe best to move this entire block/logic in the user service?

File: workflow-service/src/main/java/com/redhat/parodos/security/LoginController.java [hunk 2]
```diff
@@ -39,13 +44,25 @@
...
+		if (username != null && userService.findAllUserEntitiesByUsernameIn(List.of(username)).isEmpty()) {
+			userService.save(User.builder().username(username).enabled(true).build());
```
Comment (anludke) [inline workflow-service/src/main/java/com/redhat/parodos/security/LoginController.java:63]: Can we get additional info like first name, last name and email?

Comment (RichardW98) [reply workflow-service/src/main/java/com/redhat/parodos/security/LoginController.java:63]: I can add that from ldap for now

File: workflow-service/src/main/java/com/redhat/parodos/security/SecurityUtils.java [hunk 3]
```diff
@@ -39,12 +41,51 @@ private SecurityUtils() {
 	 * @return username.
 	 */
 	public static String getUsername() {
+		UserDetails userDetails = (UserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+		if (userDetails != null) {
+			return userDetails.getUsername();
+		}
+		else {
```
Comment (eloycoto) [inline workflow-service/src/main/java/com/redhat/parodos/security/SecurityUtils.java:48]: this else is no need at all.

Comment (RichardW98) [reply workflow-service/src/main/java/com/redhat/parodos/security/SecurityUtils.java:48]: yes it's redundent

**Review by pkliczewski — COMMENTED** [review]
/lgtm

**Pull request merged**